src/plot_api/plot_api.js

```diff
@@ -2032,12 +2095,16 @@ function _relayout(gd, aobj) {
         }
     }
 
+    // now we've collected component edits - execute them all together
+    for(arrayStr in arrayEdits) {
+        var finished = manageArrays.editContainerArray(gd,
+            Lib.nestedProperty(layout, arrayStr), arrayEdits[arrayStr], flags);
```

Comment:
For now I'll change the name per https://github.com/plotly/plotly.js/pull/1403#discussion_r102964323 -> https://github.com/plotly/plotly.js/pull/1403/commits/d29cbae52654f1e74d29e77dbc353d021353eb74 but I suspect in implementing #1412 we will restructure it along the lines you're suggesting.